### PR TITLE
Upgrade to Chef 14

### PR DESCRIPTION
**Pull request by markan**
Upgrade to chef 14, and clean up some residual issues left from the original Chef 13 work.

This picks up some work that was done in PR #1327 but didn't get included when we actually upgraded to chef 13.

### CODE REVIEW

File: omnibus/files/private-chef-cookbooks/private-chef/resources/chef_run.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+provides :chef_run
+
+property :run_list, :kind_of => String, :name_attribute => true
```
Comment (tas50) [inline omnibus/files/private-chef-cookbooks/private-chef/resources/chef_run.rb]: Get rid of :kind_of =>
Also name_attribute should be name_property

File: omnibus/files/private-chef-cookbooks/private-chef/resources/chef_run.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+provides :chef_run
```
Comment (stevendanna) [inline omnibus/files/private-chef-cookbooks/private-chef/resources/chef_run.rb]: We might want to explicitly test if this thing still works manually.  It is pretty tightly coupled to chef-client internals....

Comment (markan) [reply omnibus/files/private-chef-cookbooks/private-chef/resources/chef_run.rb]: Testing a plugin install should be sufficient, right?

**Comment by markan** [comment]
This closes #1327 (much of it was directly lifted from there, in fact)

**Comment by markan** [comment]
It also closes #265

**Pull request merged**